src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -79,9 +81,31 @@ public RuntimeConfig GetConfig()
                 subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
         }
 
+        CheckForAndSetupConfigFileWatcher();
```

Comment:
I thought about adding `Try` to the method name, but this implies that it will return a bool and I wanted to keep it void. I suppose we can return a bool that isn't being used here for clarity to the method name.

Reply:
To align with your wanting to keep it void, consider whether number 1 would make more sense. The downside I mention may not necessarily be a downside, and the method name `SetupConfigFileWatcher()` clearly communicates what side effects occur as a result of calling it and is functionally cohesive (Code Complete Ch7.2) 

I only called this out because when I read `CheckForAndSetupConfigFileWatcher()`, I couldn't confidently answer these questions:
1. What happens when `checkFor` either finds or doesn't find a configFileWatcher? Is it finding or not finding that cause positive/negative impacts or neither?
2. Is the method always guaranteed to succeed?  there's no try/catch wrapping this method call so the side effect the method name conveys is successfully applied?

Reply:
I think returning a bool makes sense because it can immediately let the caller know whether or not they have just instantiated a file watcher with the call or not. There is no need to handle the returned bool if the caller does not have a need to do so and the function name is easiest to understand.